polaris-service/src/main/java/org/apache/polaris/service/ratelimiter/RateLimiter.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.ratelimiter;
+
+/** Interface for rate limiting requests */
+public interface RateLimiter {
+  /**
+   * This signifies that a request is being made. That is, the rate limiter should count the request
+   * at this point.
+   *
+   * @return Whether the request is allowed to proceed by the rate limiter
+   */
+  boolean tryAcquire();
```

Comment:
Yes, those would be useful features that I hadn't considered.

I wonder if it makes more sense long term to expose those through the [CallContext](https://github.com/apache/polaris/blob/main/polaris-core/src/main/java/org/apache/polaris/core/context/CallContext.java), which is essentially a context corresponding to the current request which is fetchable from anywhere. It doesn't have that information today, but it easily could, and then you could re-use that information everywhere rather than each component having to pass it through.

Reply:
The request can either be put into contextVariables or added as a separate field. And CallContext doesn't have to be passed around, you can call the static getter `CallContext.getCurrentContext()`: https://github.com/apache/polaris/blob/main/polaris-core/src/main/java/org/apache/polaris/core/context/CallContext.java#L58